src/os/bluestore/bluefs_types.h

```diff
@@ -281,18 +372,20 @@ struct bluefs_transaction_t {
     encode((__u8)OP_DIR_REMOVE, op_bl);
     encode(dir, op_bl);
   }
-  void op_dir_link(std::string_view dir, std::string_view file, uint64_t ino) {
+  void op_dir_link(std::string_view dir, std::string_view file, uint64_t ino, bool is_new_wal = false) {
```

Comment:
For consideration:
How about instead of adding extra param to op_dir_link / op_dir_unlink pass FileRef& file ?

Reply:
so instead of passing extra param you want to pass another type of extra param?